enhancements/sig-architecture/15-resourcebasedscheduling/README.md

```diff
@@ -0,0 +1,191 @@
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement2
+  namespace: ns1
+spec:
+  numberOfClusters: 1
+  clusterResourcePreference:
+    type: MostAllocatable
+    clusterResources:
+      - resourceName: memory
+```
+
+## Design Details
+
+### Resource data required during scheduling
+Currently the following resource data is available for both CPU and Memory in the status of ManagedCluster, which is updated by registration agent.
+- `allocatable`, sum of the `allocatable` on each node in a managed cluster;
+- `capacity`, sum of the `capacity` on each node in a managed cluster;
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: local-cluster
+spec:
+  hubAcceptsClient: true
+  leaseDurationSeconds: 60
+status:
+  allocatable:
+    cpu: '15'
+    memory: 63194Mi
+  capacity:
+    core_worker: '18'
+    cpu: '18'
+    memory: 69938Mi
+    socket_worker: '6'
+```
+In order to support other resources, like GPU, the `allocatable` and `capacity` should be included in the status of managed cluster either.
+
+### Plugin `mostallocatabletocapacityratio`
+It is a pritoritizer and scores feasible managed clusters with the process below.
+- If the placement has `ClusterResourcePreference` specified in the spec and its `Type` is `MostAllocatableToCapacityRatio`, the score of a managed cluster is the sum of the score for each resource.  
+`score = sum(resource_x_allocatable / resource_x_capacity))`  
+while  
+  - `resource_x_allocatable` is the allocatable of resource `x` on the managed cluster;  
+  - and `resource_x_capacity` is the capacity of resource `x` on the managed cluster.  
+
+- Otherwise, the score for each managed cluster is 0.
+
+Before returning the scores to the scheduler, the data should be normalized and ensure the value falls in the range between 0 and 100.  
+`normalized = (score - min(score)) * 100 / (max(score) - min(score))`  
+
+### Plugin `mostallocatable`
+It is a pritoritizer and scores feasible managed clusters with the process below.
+- If the placement has `ClusterResourcePreference` specified in the spec and its `Type` is `MostAllocatable`, the score of a managed cluster is the sum of the score for each cluster resource.  
+`score = sum((resource_x_allocatable - min(resource_x_allocatable)) / (max(resource_x_allocatable) - min(resource_x_allocatable))`  
+while
+  - `resource_x_allocatable` is the allocatable of resource `x` on the managed cluster, 
+  - `min(resource_x_allocatable)` is the minimum allocatable of resource `x` from all managed cluster.  
+  - `max(resource_x_allocatable)` is the maxmum allocatable of resource `x` from all managed cluster.  
+- Otherwise, the score for each managed cluster is 0.  
+
+Before returning the scores to the scheduler, the data should be normalized and ensure the value falls in the range between 0 and 100.  
+`normalized = (score - min(score)) * 100 / (max(score) - min(score))`
```

Comment:
I think the test cases/version upgrade etc is required. You can set it to N/A if it is not required.

I addition I would like to see some examples, and a discussion on how it works with steady/balance plugin today.

Reply:
Added.